### PR TITLE
Move to LSP4j 0.13.0

### LINKED ISSUE #1204: Move to LSP4j 0.13.0
Move to LSP4j 0.13.0

### PR DESCRIPTION
**Pull request by angelozerr**
Move to LSP4j 0.13.0

Fixes #1204

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by rgrunber** [comment]
Also, after fixing those I see :

<details>
<summary>stacktrace</summary>

```
May 24, 2022 2:36:16 P.M. org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer fireError
SEVERE: Unable to invoke no-args constructor for class org.eclipse.lsp4j.WorkspaceSymbolResolveSupportCapabilities. Registering an InstanceCreator with Gson for this type may fix this problem.
java.lang.RuntimeException: Unable to invoke no-args constructor for class org.eclipse.lsp4j.WorkspaceSymbolResolveSupportCapabilities. Registering an InstanceCreator with Gson for this type may fix this problem.
	at com.google.gson.internal.ConstructorConstructor$14.construct(ConstructorConstructor.java:228)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:212)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
	at com.google.gson.Gson.fromJson(Gson.java:963)
	at org.eclipse.lsp4j.adapters.InitializeParamsTypeAdapter.readCapabilities(InitializeParamsTypeAdapter.java:151)
	at org.eclipse.lsp4j.adapters.InitializeParamsTypeAdapter.read(InitializeParamsTypeAdapter.java:109)
	at org.eclipse.lsp4j.adapters.InitializeParamsTypeAdapter.read(InitializeParamsTypeAdapter.java:34)
	at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199)
	at com.google.gson.Gson.fromJson(Gson.java:963)
	at org.eclipse.lsp4j.jsonrpc.json.adapters.MessageTypeAdapter.fromJson(MessageTypeAdapter.java:329)
	at org.eclipse.lsp4j.jsonrpc.json.adapters.MessageTypeAdapter.parseParams(MessageTypeAdapter.java:249)
	at org.eclipse.lsp4j.jsonrpc.json.adapters.MessageTypeAdapter.read(MessageTypeAdapter.java:119)
	at org.eclipse.lsp4j.jsonrpc.json.adapters.MessageTypeAdapter.read(MessageTypeAdapter.java:55)
	at com.google.gson.Gson.fromJson(Gson.java:963)
	at org.eclipse.lsp4j.jsonrpc.json.MessageJsonHandler.parseMessage(MessageJsonHandler.java:119)
	at org.eclipse.lsp4j.jsonrpc.json.MessageJsonHandler.parseMessage(MessageJsonHandler.java:114)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:193)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.lang.Thread.run(Thread.java:833)
	at com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:704)
	at com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:202)
Caused by: java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Method.invoke(Method.java:568)
	at com.google.gson.internal.UnsafeAllocator$1.newInstance(UnsafeAllocator.java:50)
	at com.google.gson.internal.ConstructorConstructor$14.construct(ConstructorConstructor.java:225)
	... 30 more
Caused by: java.lang.IllegalArgumentException: Type org.eclipse.lsp4j.WorkspaceSymbolResolveSupportCapabilities is instantiated reflectively but was never registered. Register the type by adding "unsafeAllocated" for the type in reflect-config.json.
	at com.oracle.svm.core.graal.snippets.SubstrateAllocationSnippets.hubErrorStub(SubstrateAllocationSnippets.java:218)
	at sun.misc.Unsafe.allocateInstance(Unsafe.java:864)
	... 33 more
```
</details>

So `WorkspaceSymbolResolveSupportCapabilities` also needs adding.

File: org.eclipse.lemminx/src/main/resources/META-INF/native-image/reflect-config.json [hunk 1]
```diff
@@ -1871,6 +1870,27 @@
...
+            "parameterTypes": []
+        }]
+    },
+    {
+        "name": "org.eclipse.lsp4j.jsonrpc.json.adapters.WorkspaceSymbolResponseAdapter",
+        "methods": [{
+            "name": "<init>",
+            "parameterTypes": []
+        }]
+    },
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply org.eclipse.lemminx/src/main/resources/META-INF/native-image/reflect-config.json]: My bad,fixed.

**Comment by rgrunber** [comment]
I see the following on Theia :

```
2022-05-25T16:28:07.484Z root INFO [hosted-plugin: 39490] PLUGIN_HOST(39490): PluginManagerExtImpl/loadPlugin(/home/rgrunber/.theia/extensions/vscode-xml-0.21.0.vsix/extension/dist/extension)
2022-05-25T16:28:07.516Z root ERROR [hosted-plugin: 39490] Activating extension 'XML' failed: /home/rgrunber/.theia/extensions/vscode-xml-0.21.0.vsix/extension/dist/extension.js:2
...
(minified js)
...
SyntaxError: Unexpected token '?'
    at wrapSafe (internal/modules/cjs/loader.js:915:16)
    at Module._compile (internal/modules/cjs/loader.js:963:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Function.module._load (/home/rgrunber/git/theia/packages/plugin-ext-vscode/lib/node/plugin-vscode-init.js:69:33)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at dynamicRequire (/home/rgrunber/git/theia/packages/core/lib/node/dynamic-require.js:28:12)
    at Object.loadPlugin (/home/rgrunber/git/theia/packages/plugin-ext/lib/hosted/node/plugin-host-rpc.js:120:65)
```

Let me do some quick testing to confirm it really is this change causing this.

**Review by rgrunber — APPROVED** [review]
I've tested this LSP4J 0.13.0 change with https://github.com/redhat-developer/vscode-xml/issues/711 reverted locally. It works for me. The XML LS starts and is working. Since the LS is independent I think this change should be allowed to proceed, and any vscode-xml issue should be fixed separately.

**Pull request merged**